src/coreComponents/dataRepository/Group.hpp

```diff
@@ -335,7 +335,7 @@ class Group
     Group * const child = m_subGroups[ key ];
     GEOSX_THROW_IF( child == nullptr,
                     "Group " << getPath() << " has no child named " << key << std::endl <<
-                    getName() << " children are: " << dumpChildrenName(),
+                    ( numSubGroups()>0 ? "The children of " + getName() + " are: " + dumpChildrenNames() : getName() + " has no children." ),
```

Comment:
What do you think of putting this code into a dedicated method / function to avoid code repetition ? There are those two version (const / not const) which are duplicating this code, moreover we may re-use this output later, and that's why I put this in `dumpChildrenNames()` and `dumpWrappersNames()`.